cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -12,6 +12,102 @@ WORKER_NODE_ROOT="${CLUSTER_NAME}-worker"
 
 OPERATOR_GIT_HASH=8d3c30de8ec5a9a0c9eeb84ea0aa16ba2395cd68  # release-4.4
 
+# This function gets a command string and invoke it
+# until the command returns an empty string or until timeout
+function retry {
+  local -r tries=$1
+  local -r wait_time=$2
+  local -r action=$3
+  local -r wait_message=$4
+
+  local result=$(eval $action)
+  for i in $(seq $tries); do
+    if [[ -z $result ]] ; then
+      echo "[$i/$tries] $wait_message"
+      sleep $wait_time
+      result=$(eval $action)
+    else
+      return 0
+    fi
+  done
+
+  return 1
+}
+
+function wait_pod {
+  local namespace=$1
+  local label=$2
+
+  local -r tries=60
+  local -r wait_time=5
+
+  local -r wait_message="Waiting for pods with $label to create"
+  local -r error_message="Pods with  label $label at $namespace namespace found"
+
+  if [[ $namespace != "" ]];then
+    namespace="-n $namespace"
+  fi
+
+  if [[ $label != "" ]];then
+    label="-l $label"
+  fi
+
+  local -r action="_kubectl get pod $namespace $label -o custom-columns=NAME:.metadata.name --no-headers"
+
+  retry "$tries" "$wait_time" "$action" "$wait_message" && return  0
+  echo $error_message && return 1
+}
+
+function wait_k8s_object {
+  local -r object_type=$1
+  local -r name=$2
+  local namespace=$3
+
+  local -r tries=60
+  local -r wait_time=3
+
+  local -r wait_message="Waiting for $object_type $name"
+  local -r error_message="$object_type $name at $namespace namespace found"
+
+  if [[ $namespace != "" ]];then
+    namespace="-n $namespace"
+  fi
+
+  local -r action="_kubectl get $object_type $name $namespace -o custom-columns=NAME:.metadata.name --no-headers"
+
+  retry "$tries" "$wait_time" "$action" "$wait_message" && return 0
+  echo $error_message && return  1
+}
+
+
+function wait_for_taint_absence {
+  local -r taint=$1
+
+  local -r tries=60
+  local -r wait_time=5
+
+  local -r wait_message="Waiting for $taint taint absence"
+  local -r error_message="Taint $taint $name did not removed"
+  local -r action='[[ -z $(_kubectl get nodes -o custom-columns=taints:.spec.taints[*].effect --no-headers | grep -i $taint) ]] && echo taint-removed || echo '
```

Comment:
see please the question about inject an error here
i believe that hiccup wont retry
since it will just think there arent taints

Reply:
how should I inject an error or simulate a "hiccup"?

I checked this function like this:
I added a taint to one of the nodes and ran `wait_for_taint_absence` function (manually no through the script)
and it keep running and checking if there is a taint (according to the action passed to retry)
then in another session I removed the taint and the function returned.

Reply:
just change the "oc get" to "oc ge"
command error will play as error occur (timeout and so on)

Reply:
I tried it and the function stops immediately with an error from `kubectl, 
which is fine because `wait_for_taint_absence` is dedicated function for waiting for a taint to remove,
the `action` inside it should not be changed.

Reply:
i dont think its fine, an hiccup should have retry
hiccup means for example etcd isnt reachable
the answer you have when it happens doesnt represnt the real state

i mentioned hiccup becuase hiccup need retry
see please the PR Petr did for this reason
https://github.com/kubevirt/kubevirtci/pull/364
or see #301 
where i fixed that hiccups due to 3 masters, which takes lots of IO, would have retry

Reply:
gotcha, I will fix it.